Fix detection of several libraries

1) Use system zlib instead of embedded zlib-ng

2) Fix minizip detection

--- CMakeLists.txt.orig	2024-12-01 23:00:52 UTC
+++ CMakeLists.txt
@@ -672,10 +672,10 @@ dolphin_find_optional_system_library_pkgconfig(ZSTD li
 
 dolphin_find_optional_system_library_pkgconfig(ZSTD libzstd>=1.4.0 zstd::zstd Externals/zstd)
 
-add_subdirectory(Externals/zlib-ng)
+dolphin_find_optional_system_library_pkgconfig(ZLIB zlib ZLIB::ZLIB Externals/zlib-ng)
 
 dolphin_find_optional_system_library_pkgconfig(MINIZIP
-  "minizip>=4.0.4" minizip::minizip Externals/minizip-ng
+  "minizip-ng>=4.0.4" minizip::minizip Externals/minizip-ng
 )
 
 dolphin_find_optional_system_library(LZO Externals/LZO)
